omero_fpbioimage/views.py

```diff
@@ -26,8 +49,13 @@ def fpbioimage_png(request, image_id, the_z, conn=None, **kwargs):
     """Render png for image at specified Z section."""
     image = conn.getObject('image', image_id)
     jpeg_data = image.renderJpeg(the_z, 0, compression=0.9)
-
+    width = max(image.getSizeX, image.getSizeY)
+    if width > 500:
+        width = 500
     i = Image.open(StringIO(jpeg_data))
+    p = (width/float(i.size[0]))
+    hsize = int((float(i.size[1])*float(p)))
+    i = i.resize((width, hsize), PIL.Image.ANTIALIAS)
```

Comment:
If sizeY is greater than sizeX then sizeX remains at 500, or unchanged if less than 500.
Also, p can be greater than 1 (scale up).
E.g. I have an image of ```125 x 484``` and it's being scaled by ```p 3.872``` to ```484 x 1874```.